src/main.rs

```diff
@@ -283,7 +283,12 @@ fn main() {
         let working_dir = tmp_path.join(format!("{}", i));
         let source_root = source_root.clone();
 
-        let t = thread::spawn(move || {
+        let mut  thread_name = String::from("Consumer ");
+        thread_name.push_str(&i.to_string());
+
+        let t = thread::Builder::new().name(thread_name);
+        
+        let join_handle_t = t.spawn(move || {
```

Comment:
Try to see if you can do `thread::Builder::new().name(thread_name).spawn` instead

Reply:
I was doing this before and join() wasn't working. I understood that threads created with Builder cannot join().

Reply:
I think they can, it's just that you needed an `unwrap` at the end of `spawn` to actually get a JoinHandle object, otherwise you get a [Result](https://doc.rust-lang.org/std/result/) object.

Reply:
Done!